cg/meta/workflow/balsamic.py

```diff
@@ -431,12 +434,57 @@ def get_normal_sample_name(self, case_id: str) -> Optional[str]:
         if sample_obj:
             return sample_obj.internal_id
 
+    def get_latest_observations(self, observations_pattern: str) -> Optional[str]:
+        """
+        Returns the latest loqusdb dump file (loqusdb_<observations>_export-<date>-.vcf.gz) matching an
+        observations pattern.
+        """
+
+        available_files: iter = Path(self.loqusdb_path).glob(f"*{observations_pattern}*")
+        return get_last_element_from_iterator(available_files)
+
+    def get_verified_observations(
+        self,
+        clinical_observations: List[str],
+        cancer_observations: List[str],
+    ) -> dict:
+        """Returns verified clinical and cancer observations paths."""
+
+        # Clinical observations
+        clinical_snv_observations: str = get_string_from_list_by_pattern(
+            ObservationFileWildcards.CLINICAL_SNV, clinical_observations
+        ) or self.get_latest_observations(ObservationFileWildcards.CLINICAL_SNV)
+        clinical_sv_observations: str = get_string_from_list_by_pattern(
+            ObservationFileWildcards.CLINICAL_SV, clinical_observations
+        ) or self.get_latest_observations(ObservationFileWildcards.CLINICAL_SV)
+
+        # Cancer observations
+        cancer_all_snv_observations: str = get_string_from_list_by_pattern(
+            ObservationFileWildcards.CANCER_ALL_SNV, cancer_observations
+        ) or self.get_latest_observations(ObservationFileWildcards.CANCER_ALL_SNV)
+        cancer_somatic_snv_observations: str = get_string_from_list_by_pattern(
+            ObservationFileWildcards.CANCER_SOMATIC_SNV, cancer_observations
+        ) or self.get_latest_observations(ObservationFileWildcards.CANCER_SOMATIC_SNV)
+        cancer_somatic_sv_observations: str = get_string_from_list_by_pattern(
+            ObservationFileWildcards.CANCER_SOMATIC_SV, cancer_observations
+        ) or self.get_latest_observations(ObservationFileWildcards.CANCER_SOMATIC_SV)
```

Comment:
If you make a separate function with argument for taking in: `ObservationFileWildcards` and `observations`. Then you can generate the dict, below by calling the function with the appropriate args. You should even be able to make it loop over a Dict[list of "keys", Dict[ObservationFileWildcards, observations]] and update the value in the dict by calling the function over each list of "keys". I hope that made sense - let me know otherwise 😄